internal/configs/transportserver_test.go

```diff
@@ -345,6 +350,61 @@ func TestGenerateUnixSocket(t *testing.T) {
 	}
 }
 
+func TestGenerateTransportServerHealthChecks(t *testing.T) {
+	tests := []struct {
+		healthCheck *conf_v1alpha1.HealthCheck
+		expected    *version2.StreamHealthCheck
+		msg         string
+	}{
+		{
+			healthCheck: &conf_v1alpha1.HealthCheck{
+				Enabled:  false,
+				Timeout:  "30s",
+				Jitter:   "30s",
+				Port:     80,
+				Interval: "20s",
+				Passes:   4,
+				Fails:    5,
+			},
+			expected: nil,
+			msg:      "health checks disabled",
+		},
+		{
+			healthCheck: &conf_v1alpha1.HealthCheck{},
+			expected:    nil,
+			msg:         "nil health checks",
+		},
+		{
+			healthCheck: &conf_v1alpha1.HealthCheck{
+				Enabled:  true,
+				Timeout:  "30s",
+				Jitter:   "30s",
+				Port:     80,
+				Interval: "20s",
+				Passes:   4,
+				Fails:    5,
+			},
+			expected: &version2.StreamHealthCheck{
+				Enabled:  true,
+				Timeout:  "30s",
+				Jitter:   "30s",
+				Port:     80,
+				Interval: "20s",
+				Passes:   4,
+				Fails:    5,
+			},
+			msg: "valid health checks",
+		},
+	}
+
+	for _, test := range tests {
+		result := generateTransportServerHealthCheck(test.healthCheck)
+		if diff := cmp.Diff(result, test.expected); diff != "" {
+			t.Errorf("generateTransportServerHealthCheck() returned %v but expected %v", result, test.expected)
```

Comment:
could you print the diff here, so that it is clear which fields are not equal?

An example from a similar test:
```
		if diff := cmp.Diff(test.expected, result); diff != "" {
			t.Errorf("generatePolicies() '%v' mismatch (-want +got):\n%s", test.msg, diff)
		}
```